examples/Azure_IoT_Central_ESP32/AzureIoT.cpp

```diff
@@ -0,0 +1,1258 @@
+  EXIT_IF_AZ_FAILED(azrc, RESULT_ERROR, "Failed to get the commands response topic");
+
+  mqtt_message.topic = az_span_slice(mqtt_message.topic, 0, topic_length);
+  mqtt_message.payload = payload;
+  mqtt_message.qos = mqtt_qos_at_most_once;
+
+  packet_id = azure_iot->config->mqtt_client_interface.mqtt_client_publish(azure_iot->mqtt_client_handle, &mqtt_message);
+  
+  if (packet_id < 0)
+  {
+    azure_iot->state = azure_iot_state_error;
+    LogError("Failed publishing command response (%.*s).", az_span_size(request_id), az_span_ptr(request_id));
+    return RESULT_ERROR;
+  }
+  else
+  {
+    return RESULT_OK;
+  }
+}
+
+/* --- Implementation of internal functions --- */
+
+/*
+ * @brief           Gets the number of seconds since UNIX epoch until now.
+ * @return uint32_t Number of seconds.
+ */
+static uint32_t get_current_unix_time()
+{
+  time_t now = time(NULL);
+  return (now == INDEFINITE_TIME ? 0 : (uint32_t)(now));
+}
+
+/*
+ * @brief           Initializes the Device Provisioning client and generates the config for a MQTT client.
+ * @param[in]       azure_iot          A pointer to an initialized instance of azure_iot_t.
+ * @param[in]       mqtt_client_config A pointer to a generic structure to contain the configuration for
+ *                                     creating and connecting an MQTT client to Azure Device Provisioning service.
+ * 
+ * @return int      0 on success, non-zero if any failure occurs.
+ */
+static int get_mqtt_client_config_for_dps(azure_iot_t* azure_iot, mqtt_client_config_t* mqtt_client_config)
+{
+  az_span data_buffer_span, client_id_span, username_span, password_span;
+  size_t client_id_length, username_length, password_length;
+  az_result azrc;
+
+  azrc = az_iot_provisioning_client_init(
+      &azure_iot->dps_client,
+      AZ_SPAN_FROM_STR(DPS_GLOBAL_ENDPOINT_MQTT_URI_WITH_PORT),
+      azure_iot->config->dps_id_scope,
+      azure_iot->config->dps_registration_id,
+      NULL);
+  EXIT_IF_AZ_FAILED(azrc, RESULT_ERROR, "Failed to initialize provisioning client.");
+
+  data_buffer_span = azure_iot->data_buffer;
+  
+  password_span = az_span_split(data_buffer_span, MQTT_PASSWORD_BUFFER_SIZE, &data_buffer_span);
+  EXIT_IF_TRUE(az_span_is_empty(password_span), RESULT_ERROR, "Failed allocating buffer for password_span.");
+
+  password_length = generate_sas_token_for_dps(
+    &azure_iot->dps_client,
+    azure_iot->config->device_key,
+    azure_iot->config->sas_token_lifetime_in_minutes,
+    data_buffer_span,
+    azure_iot->config->data_manipulation_functions,
+    password_span,
+    &azure_iot->sas_token_expiration_time);
+  EXIT_IF_TRUE(password_length == 0, RESULT_ERROR, "Failed creating mqtt password for DPS connection.");    
+
+  client_id_span = az_span_split(data_buffer_span, MQTT_CLIENT_ID_BUFFER_SIZE, &data_buffer_span);
+  EXIT_IF_TRUE(az_span_is_empty(client_id_span), RESULT_ERROR, "Failed allocating buffer for client_id_span.");
```

Comment:
avoid using 'allocating' since you are not really allocating memory. 

```suggestion
  EXIT_IF_TRUE(az_span_is_empty(client_id_span), RESULT_ERROR, "Failed parsing buffer for client_id_span.");
```

Reply:
Fine. Renamed all "allocating" to "reserving"